File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/fragment/entitieslist/core/trashlist/TrashListPresenter.java
package com.github.android.lvrn.lvrnproject.view.fragment.entitieslist.core.trashlist;

import com.github.android.lvrn.lvrnproject.persistent.entity.Note;
import com.github.android.lvrn.lvrnproject.service.form.NoteForm;
import com.github.android.lvrn.lvrnproject.view.fragment.entitieslist.EntitiesListWithSearchPresenter;

/**
 * @author Andrii Bei <psihey1@gmail.com>
 */

public interface TrashListPresenter extends EntitiesListWithSearchPresenter<Note, NoteForm> {

    void removeNoteForever();

    void restoreNote();
}

File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/fragment/entitieslist/core/trashlist/TrashListPresenter.kt
package com.github.android.lvrn.lvrnproject.view.fragment.entitieslist.core.trashlist

import com.github.android.lvrn.lvrnproject.service.form.NoteForm
import com.github.android.lvrn.lvrnproject.view.fragment.entitieslist.EntitiesListWithSearchPresenter
import com.github.valhallalabs.laverna.persistent.entity.Note

interface TrashListPresenter : EntitiesListWithSearchPresenter<Note, NoteForm> {

    fun removeNoteForever(position: Int)

    fun restoreNote(position: Int)
}